queries/sun_position.py

```diff
@@ -0,0 +1,673 @@
+_SOLAR_DICT_TYPE = Dict[
+    str, Dict[Union[str, datetime.date], Union[LatLonTuple, _SOLAR_ROW_TYPE]]
+]
+
+_SOLAR_ENUM_TO_WORD: Dict[_SOLAR_POS_ENUM, str] = {
+    _SOLAR_POSITIONS.MIÐNÆTTI: "Miðnætti",
+    _SOLAR_POSITIONS.DÖGUN: "Dögun",
+    _SOLAR_POSITIONS.BIRTING: "Birting",
+    _SOLAR_POSITIONS.SÓLRIS: "Sólris",
+    _SOLAR_POSITIONS.HÁDEGI: "Hádegi",
+    _SOLAR_POSITIONS.SÓLARLAG: "Sólarlag",
+    _SOLAR_POSITIONS.MYRKUR: "Myrkur",
+    _SOLAR_POSITIONS.DAGSETUR: "Dagsetur",
+    _SOLAR_POSITIONS.SÓLARHÆÐ: "Sólarhæð",
+}
+
+
+def QSunQuery(node: Node, params: QueryStateDict, result: Result) -> None:
+    # Set the query type
+    result.qtype = _SUN_QTYPE
+
+
+def QSunIsWillWas(node: Node, params: QueryStateDict, result: Result) -> None:
+    if result._nominative == "verður":
+        result["will_be"] = True
+
+
+### QSunPositions ###
+
+
+def QSunMiðnætti(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.MIÐNÆTTI
+
+
+def QSunDögun(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.DÖGUN
+
+
+def QSunBirting(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.BIRTING
+
+
+def QSunSólris(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.SÓLRIS
+
+
+def QSunHádegi(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.HÁDEGI
+
+
+def QSunSólarlag(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.SÓLARLAG
+
+
+def QSunMyrkur(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.MYRKUR
+
+
+def QSunDagsetur(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.DAGSETUR
+
+
+def QSunSólarhæð(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["solar_position"] = _SOLAR_POSITIONS.SÓLARHÆÐ
+
+
+### QSunDates ###
+
+
+def QSunToday(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["date"] = datetime.date.today()
+
+
+def QSunYesterday(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["date"] = datetime.date.today() - datetime.timedelta(days=1)
+
+
+def QSunTomorrow(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["date"] = datetime.date.today() + datetime.timedelta(days=1)
+
+
+### QSunLocation ###
+
+
+def QSunCapitalRegion(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["city"] = "Reykjavík"
+
+
+def QSunArbitraryLocation(node: Node, params: QueryStateDict, result: Result) -> None:
+    result["city"] = capitalize_placename(result._nominative)
+
+
+###
+
+_ALMANAK_HI_URL: str = "http://www.almanak.hi.is/solgang.html"
```

Comment:
Are you sure there's no nice API for this?

Reply:
I only found an XML API from vedur.is which didn't quite offer the information that I wanted.